calabash-cucumber/lib/calabash-cucumber/usage_tracker.rb

```diff
@@ -56,6 +57,21 @@ def post_usage_async
       private
 
       # @!visibility private
+      def preferences
+        @preferences ||= Calabash::Cucumber::Preferences.new
+      end
+
+      # @!visibility private
+      def user_id
+        @user_id ||= preferences.user_id
+      end
+
+      # @!visibility private
+      def allowed_to_track
+        @allowed_to_track ||= preferences.usage_tracking
+      end
+
+      # @!visibility private
       def self.track_usage?
         @@track_usage && !self.xtc?
```

Comment:
> Specifically, @@track_usage is class variable that acts as a (class) global on/off switch.
> Savvy users can also use this variable to control usage-tracking on a per project basis.

I guess my thinking is that savvy users could just manipulate the actual JSON file to make this switch if they _really_ needed to. I prefer to avoid having two switches that perform the same function - IMO it leads to some unnecessary complexity. 

```ruby
if @@track_usage && @allowed_to_track { }
```
is semantically equivalent to
```ruby
if @@user_has_enabled_usage_tracking && @user_has_enabled_usage_tracking { }
```
The only appreciable difference is that one occurs in memory and the other on disk.